src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpSystemProxy.cs

```diff
@@ -60,26 +66,86 @@ private HttpSystemProxy(WinInetProxyHelper proxyHelper, SafeWinHttpHandle sessio
 
                 if (!string.IsNullOrWhiteSpace(proxyHelper.ProxyBypass))
                 {
+                    int idx = 0;
+                    int start = 0;
+                    string tmp;
+
                     // Process bypass list for manual setting.
-                    string[] list = proxyHelper.ProxyBypass.Split(';');
-                    List<string> tmpList = new List<string>();
+                    _bypass = new List<Regex>(proxyHelper.ProxyBypass.Length / 5); // Best guess based on string length.
```

Comment:
Where does the 5 come from?  This is a bit strange; it means, for example, that a Length of 5 will end up allocating a list with an underlying array of 1 element (5/4 == 1), but a Length of 3 will end up allocating a list without an underlying array that'll then immediately grow to 4 elements once the first item is added.

Reply:
this _is_ magic number assuming each name is at least 5 characters on average. (this no longer list of splits) I could leave it empty but comments from @pjanotti suggested that having bigger initial size is cheaper than reallocation. 
To explain this I added better comment.